### ISSUE TITLE
NullPointerException on decrypting an empty stream

### ISSUE THREAD

**Issue by sa-sh** [LibrePDF/OpenPDF#443]
**Describe the bug**
NullPointerException occurs at com.lowagie.text.pdf.StandardDecryption#finish(StandardDecryption.java:105)
in PDF enctypted with AES while decrypting object of empty stream (**cipher** is not initialized in this case)

sample pdf object that causes NPE if try to modify (using PDFStamper) encrypted PDF with such object

> 41 0 obj
> <</Length 0/Type/Pattern/PatternType 1/PaintType 1/TilingType 2/BBox[0 0 10 10]/XStep 10/YStep 10/Resources 42 0 R>>stream
> 
> endstream
> endobj

stacktrace of error

```java
java.lang.NullPointerException
	at com.lowagie.text.pdf.StandardDecryption.finish(StandardDecryption.java:105)
	at com.lowagie.text.pdf.PdfEncryption.decryptByteArray(PdfEncryption.java:582)
	at com.lowagie.text.pdf.PdfReader.getStreamBytesRaw(PdfReader.java:2355)
	at com.lowagie.text.pdf.PdfReader.getStreamBytesRaw(PdfReader.java:2385)
	at com.lowagie.text.pdf.PRStream.toPdf(PRStream.java:222)
	at com.lowagie.text.pdf.PdfIndirectObject.writeTo(PdfIndirectObject.java:164)
	at com.lowagie.text.pdf.PdfWriter$PdfBody.add(PdfWriter.java:410)
	at com.lowagie.text.pdf.PdfWriter.addToBody(PdfWriter.java:841)
	at com.lowagie.text.pdf.PdfStamperImp.close(PdfStamperImp.java:320)
	at com.lowagie.text.pdf.PdfStamper.close(PdfStamper.java:246)
```

**To Reproduce**
1. Sample Code

```java
      try (PdfReader reader = new PdfReader("aes-128-document.pdf")) {
         try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
            PdfStamper stamper = new PdfStamper(reader, out);
            // do pdf stuff
            stamper.close();  // NPE here
         }
      }
```

2. Unit-Test

```java
    @Test
    public void test() {
       // setup AES 128 encryption
       PdfEncryption decrypt = new PdfEncryption();
       decrypt.setCryptoMode(PdfWriter.ENCRYPTION_AES_128, 0);
       byte[] key = new byte[16]; // create fake key, it does not matter for this test
       decrypt.setupByEncryptionKey(key, 128);
       decrypt.setHashKey(0, 0);
       // try to decrypt an empty array
       byte[] result = decrypt.decryptByteArray(new byte[] {});
       // empty array expected in result and no exception
       Assert.assertNotNull(result);
       Assert.assertEquals(result.length, 0);
    }
```

**Expected behavior**
no exception

**Screenshots**

**Additional context**

itext 5 has fixed this bug by simple checking for null chiper in StandardDecryption#finish method

**Comment by andreasrosdal** [comment]
pull requests welcome

**Comment by sa-sh** [comment]
thanks, request #445 opened

**Comment by asturio** [comment]
Your PR is now merged. I'll close the issue, and create a new release in the next days.

**Comment by asturio** [comment]
Resolved with #445
